### PR TITLE
InetSocketAddress support

### PR DESCRIPTION
**Pull request by vdantu**
*Issue #, if available:*

*Description of changes:*
Added support to use AF_INET and AF_UNIX for backend worker

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: frontend/server/src/main/java/com/amazonaws/ml/mms/wlm/WorkerLifeCycle.java [hunk 1]
```diff
@@ -38,10 +40,22 @@ public WorkerLifeCycle(ConfigManager configManager) {
...
+        SocketAddress address = NettyUtils.getSocketAddress(port);
+        String[] args = new String[6];
         args[0] = "python";
         args[1] = "mms/model_service_worker.py";
-        args[2] = NettyUtils.getSocketAddress(port).toString();
+        args[2] = "--sock-name";
+        args[4] = "--sock-type";
+        if (address instanceof InetSocketAddress) {
+            args[5] = "tcp";
+            args[3] =
```
Comment (frankfliu) [inline frontend/server/src/main/java/com/amazonaws/ml/mms/wlm/WorkerLifeCycle.java]: It's better only pass port, let backend decide witch interface to bind on.

File: frontend/server/src/main/java/com/amazonaws/ml/mms/wlm/WorkerLifeCycle.java [hunk 2]
```diff
@@ -38,10 +40,22 @@ public WorkerLifeCycle(ConfigManager configManager) {
...
+                    ((InetSocketAddress) address).getHostString()
+                            + ":"
+                            + ((InetSocketAddress) address).getPort();
+        } else {
+            args[5] = "unix";
+            args[3] = address.toString();
```
Comment (frankfliu) [inline frontend/server/src/main/java/com/amazonaws/ml/mms/wlm/WorkerLifeCycle.java]: use address.toString() was a hack, better get the uds path directly.

Comment (vdantu) [reply frontend/server/src/main/java/com/amazonaws/ml/mms/wlm/WorkerLifeCycle.java]: Addressed this.

File: frontend/server/src/main/java/com/amazonaws/ml/mms/wlm/WorkerLifeCycle.java (hunk 1 above)
Comment (vdantu) [reply frontend/server/src/main/java/com/amazonaws/ml/mms/wlm/WorkerLifeCycle.java]: Split it into host-port on the backend

File: mms/model_service_worker.py [hunk 3]
```diff
@@ -34,30 +34,42 @@
...
-        try:
-            os.unlink(s_name)
-        except OSError:
-            if os.path.exists(s_name):
-                raise MMSError(err.SOCKET_ERROR, "socket already in use: {}.".format(s_name))
-
         try:
             msg = "Listening on port: {}\n".format(s_name)
             log_msg(msg)
+            socket_family = socket.AF_UNIX
```
Comment (frankfliu) [inline mms/model_service_worker.py]: socket_family = socket.AF_UNIX if s_type == 'unix' else socket.AF_INET

**Comment by vdantu** [comment]
@frankfliu , @vrakesh . Could I get a review for this PR?

**Pull request merged**